### ISSUE TITLE
building on JDK 8

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#1399]
## steps

try building on JDK 8
## problems

``` scala
[error] sbt/compile/interface/src/main/scala/xsbt/ExtractAPI.scala:479: java.util.Comparator does not take type parameters
[error]   private[this] val sortClasses = new Comparator[Symbol] {
[error]                                       ^
```
## expectations

it works.

**Comment by jsuereth** [comment]
I think missed the most important failure:

```
[error] error while loading CharSequence, class file '/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar(java/lang/CharSequence.class)' is broken
[error] (bad constant pool tag 15 at byte 1501)
[error] error while loading AnnotatedElement, class file '/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar(java/lang/reflect/AnnotatedElement.class)' is broken
[error] (bad constant pool tag 15 at byte 2713)
```

unless you don't see that.

**Comment by jsuereth** [comment]
Oh, also, scala prior to 2.10.4 didn't support JDK8, so we can't either for the old compiler interfaces precompiled.   SO, this is related to that.  Drop your precompiled Scala projects and you amy see it start working.

**Comment by jsuereth** [comment]
Final note on failure, see the fialed projectss:

```
[error] (precompiled-2_8_2/compile:compile) Compilation failed
[error] (precompiled-2_9_3/compile:compile) Compilation failed
[error] (precompiled-2_9_2/compile:compile) Compilation failed
```

**Comment by dansanduleac** [comment]
I should note that building sbt -- well, at least the `testing/agent` project -- on anything greater than jdk 1.6 results in not being able to fork tests with an older jdk than that which was used to compile sbt, and looks like (note how the tests even _succeed_ in this case):

```
Internal error when running tests: java.lang.UnsupportedClassVersionError: sbt/ForkTags : Unsupported major.minor version 51.0
Exception in thread "Thread-10" java.io.EOFException
        at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2598)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1318)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:370)
        at sbt.React.react(ForkTests.scala:114)
        at sbt.ForkTests$$anonfun$mainTestTask$1$Acceptor$2$.run(ForkTests.scala:74)
        at java.lang.Thread.run(Thread.java:744)
[success] Total time: 8 s, completed 30-Jun-2014 22:42:04
```

**Comment by Blaisorblade** [comment]
@dansanduleac

> I should note that building sbt -- well, at least the testing/agent project -- on anything greater than jdk 1.6 results in not being able to fork tests with an older jdk than that which was used to compile sbt

The error suggests that forked tests still run SBT code. Code compiled with JDK 1.X is not guaranteed to work on JDK 1.Y if Y < X, unless you use the -target switch of javac/scalac. major.minor = 51.0 means that the code is targeted to Java 1.7. Details for instance at http://stackoverflow.com/a/11432195/53974.

**Comment by NightRa** [comment]
Same error now, Scala 2.11.7, JDK 1.8.60.

**Comment by jasongoodwin** [comment]
updating to 2.10.4 resolved comparator issue for me

**Comment by humoyun** [comment]
updating to 2.10.4 resolved comparator issue for me too

**Comment by dwijnand** [comment]
At least for sbt/sbt, fixed in #2771
